core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -391,8 +393,17 @@ public static Wallet fromWatchingKey(NetworkParameters params, DeterministicKey
     /**
      * Creates a wallet that tracks payments to and from the HD key hierarchy rooted by the given watching key. The
      * account path is specified. The key is specified in base58 notation and the creation time of the key. If you don't
-     * know the creation time, you can pass {@link DeterministicHierarchy#BIP32_STANDARDISATION_TIME_SECS}.
+     * know the creation time, you can pass empty and it will use {@link DeterministicHierarchy#BIP32_STANDARDISATION_TIME_SECS}.
      */
+    public static Wallet fromWatchingKeyB58(NetworkParameters params, String watchKeyB58, Optional<Instant> creationTime) {
+        if (creationTime.isPresent())
+            return fromWatchingKeyB58(params, watchKeyB58, creationTime.get().getEpochSecond());
```

Comment:
Isn't this calling a deprecated method?

Reply:
Yes, I decided to implement it that way around because it was easier.

Reply:
I generally prefer to implement the deprecated methods with the non-deprecated methods for the following reasons:

1. It makes things clear -- at the level of the code -- what the canonical way of doing things is.
2. It eliminates deprecated warnings in compiles and in the IDE
3. It makes it easier to remove the deprecated method in the future.

I'll address this in my attempt at an alternative commit.